sdm_versions_manager/initial_population.py

```diff
@@ -0,0 +1,191 @@
+# A script for populating the database with existing data model versions based on commit history.
+
+# This script is designed to run independently. 
+# It will take the name of the data model and subject as inputs and fetch the historical commit data 
+# from the GitHub repository to populate the database with past versions. 
+# This script can utilize the GitHub API to retrieve commit history and extract relevant information.
+
+
+# SDMs versions data structure to fill the dataset with:
+#json_data_db = {
+#    "subject": "",
+#    "dataModel": "",
+#    "version": "",
+#    "schemaLink": "",
+#    "commitDate"
+#    "commitHash": "",
+#}
+
+import os
+import re
+import json
+import requests
+import logging
+from dotenv import load_dotenv
+from datetime import datetime
+from database import insert_data_to_mongo
+
+
+# Load environment variables from .env file
+load_dotenv()
+
+
+# Logging setup
+os.makedirs('logs', exist_ok=True)
+
+logging.basicConfig(filename='logs/initial_population.log', level=logging.INFO,
+                    format='%(asctime)s - %(message)s', datefmt='%Y-%m-%d %H:%M:%S')
+
+
+# Constants for GitHub API
+GITHUB_BASE_URL = "https://github.com/smart-data-models"
+GITHUB_API_URL = "https://api.github.com/repos/smart-data-models"
+HEADERS = {
+    "Authorization": f"token {os.getenv('GITHUB_TOKEN')}",
+    "Accept": "application/vnd.github.v3+json"
+}
+
+def load_config(config_file) -> dict:
+    """Load configuration from a JSON file containing the data models and subjects."""
+    
+    with open(config_file, 'r') as file:
+        return json.load(file)
+
+
+def construct_schema_link(subject, data_model):
+    """Construct the GitHub link to the schema.json file."""
+
+    base_url = "https://github.com/smart-data-models"
+    repo_name = f"dataModel.{subject}"
+    return f"{base_url}/{repo_name}/blob/master/{data_model}/schema.json"
+
+
+def get_commits_from_github(subject, data_model):
+    """Fetch commit history from GitHub for a data model."""
+
+    repo_name = f"dataModel.{subject}"
+    url = f"{GITHUB_API_URL}/{repo_name}/commits?path={data_model}/schema.json"
+    
+    all_commits = []
+    page = 1
+
+    while True:
```

Comment:
Tech. Debt, It is working only if the number of queries to GitHub is bellow a certain number, if you increase this number, the queries to GitHub do not work and you need to wait 1h (Tech. Debt).